src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -127,19 +127,29 @@ PathTuples joinPredicateAndObject(VarOrPath predicate, ObjectList objectList) {
   return tuples;
 }
 
+namespace {
+// Converts the PrefixMap to the legacy data format used by ParsedQuery
+vector<SparqlPrefix> convertPrefixMap(
+    const SparqlQleverVisitor::PrefixMap& map) {
+  vector<SparqlPrefix> prefixes;
+  for (auto const& [label, iri] : map) {
+    prefixes.emplace_back(label, iri);
+  }
+  return prefixes;
+}
+}  // namespace
+
 // ____________________________________________________________________________________
-std::variant<ParsedQuery, Triples> Visitor::visitTypesafe(
-    Parser::QueryContext* ctx) {
+ParsedQuery Visitor::visitTypesafe(Parser::QueryContext* ctx) {
   // The prologue (BASE and PREFIX declarations)  only affects the internal
   // state of the visitor.
   visitTypesafe(ctx->prologue());
-  if (ctx->selectQuery()) {
-    return visitTypesafe(ctx->selectQuery());
-  }
-  if (ctx->constructQuery()) {
-    return visitTypesafe(ctx->constructQuery());
-  }
-  reportError(ctx, "QLever only supports select and construct queries");
+  auto query =
+      visitAlternative<ParsedQuery>(ctx->selectQuery(), ctx->constructQuery());
+  query._originalString = ctx->getText();
+  query._prefixes = convertPrefixMap(_prefixMap);
+
+  return query;
```

Comment:
Unfortunately in this case `visitAlternative` as is is not correct, as the else case is not `AD_FAIL` but "your query is perfectly valid but not supported by QLever", so we want to keep the error message from the old code (maybe with capitalized `SELECT` and `CONSTRUCT`.